### PR TITLE
[fixes #48] Harmonize Angular and React Starters

### LINKED ISSUE #48: Harmonize angular and react starters
Updates to run scripts:
- no more webpack-dev-server: instead devmode runs `nodemon` and applies webpack middleware to the express app.  This gives a 'watch' mode that works for changes to both client and server code.
- fixed npm start to actually serve all static assets, not just `index.html`
- misc tweaks to harmonize the webpack config as much as possible with the angular starter.

### PR DESCRIPTION
**Pull request by SethDavenport**

### CODE REVIEW

**Comment by SethDavenport** [comment]
See issue #48 

@bennett000 @neilff

File: server.js [hunk 1]
```diff
@@ -22,9 +22,7 @@ if (process.env.NODE_ENV !== 'production') {
   app.use(require('webpack-hot-middleware')(compiler));
 }
 
-app.get('/', (req, res) => {
-  res.sendFile(path.join(__dirname, '/dist/index.html'));
-});
+app.get('*', express.static(path.join(__dirname, 'dist')));
```
Comment (xorgy) [inline server.js]: Will this work for routes below root?

Comment (SethDavenport) [inline server.js]: It should - what do you suggest instead? `app.use('/', express.static(...))`?

**Comment by SethDavenport** [comment]
Actually leaning more towards something like this for serving the client from node: https://github.com/SethDavenport/sethdavenport.com/blob/master/server/serve-client.js

- in prod mode, serves statically from dist
- in dev move, serves from webpack's dev middleware (with watching)
- in both modes, supports html5mode history routing

This is what I'm using on my website; if there's interest I can spin it into it's own npm package.

**Pull request merged**